packages/grid_client/src/primitives/nodes.ts

```diff
@@ -333,9 +335,42 @@ class Nodes {
       });
   }
 
+  /**
+   * Sets the features of the node according to filter options.
+   *
+   * @param options - An object containing filter options to set the node's features.
+   * @returns A string array representing the node's features.
+   */
+  getFeaturesFromFilters(options: FilterOptions = {}): string[] {
+    const features: string[] = [];
+    if (options.publicIPs) {
+      features.push(WorkloadTypes.ipv4);
+    }
+    if (options.hasIPv6) {
+      features.push(WorkloadTypes.ip);
+    }
+    if (options.wireguard) {
+      features.push("wireguard");
+    }
+    if (options.mycelium) {
+      features.push("mycelium");
+    }
+    if (options.planetary) {
+      features.push("yggdrasil");
+    }
+
+    if (options.gateway) {
```

Comment:
we need to veriffy if this is valid using custom domain name or not

Reply:
now we have Feature enum, I suggest to use switch case and include the rest of the features

Reply:
As per discussion with @0oM4R and @AhmedHanafy725, we will not include the rest of the features in the function as they're supported by both zos4 and zos3 and won't have an affect